src/service_router.rs

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use context::{ ContextTrait, Shared };
+use iron::{Request, Response, IronResult};
+use iron::headers::ContentType;
+use iron::status::Status;
+use router::Router;
+use core::marker::Reflect;
+
+
+pub fn generate<Ctx: Send + Reflect + ContextTrait + 'static> (context: Shared<Ctx>) -> Router {
+
+    let mut router = Router::new();
+    let context1 = context.clone();
+    router.get("list.json", move |_: &mut Request| -> IronResult<Response> {
+        // Build a json representation of the services.
+        let ctx = context1.lock().unwrap();
+        let serialized = itry!(ctx.services_as_json());
+
+        let mut response = Response::with(serialized);
+        response.status = Some(Status::Ok);
+        response.headers.set(ContentType::json());
+
+        Ok(response)
+    });
+
+    let context2 = context.clone();
+    router.get(":service", move |req: &mut Request| -> IronResult<Response> {
```

Comment:
Why did you remove the /:command part from the url? It's used by the current DummyService at https://github.com/fxbox/foxbox/blob/master/src/dummy_adapter.rs#L74 (yeah, if we had tests...)